api/use_cases.py

```diff
@@ -122,3 +124,82 @@ def _get_subscriptions(self, topic):
         if not subscribers:
             logger.info("Nobody to notify about the topic %s", topic)
         return subscribers
+
+
+class InvalidCallbackResponse(Exception):
+    pass
+
+
+class DeliverCallbackUseCase:
+    """
+    Is used by a callback deliverer worker
+
+    Reads queue delivery_outbox_repo consisting of tasks in format:
+        (url, message)
+
+    Then such message should be either sent to this URL and the task is deleted
+    or, in case of any error, not to be re-scheduled again
+    (up to MAX_ATTEMPTS times)
+
+    """
+
+    MAX_ATTEMPTS = 3
+
+    def __init__(self, delivery_outbox_repo: repos.DeliveryOutboxRepo, hub_url):
+        self.delivery_outbox = delivery_outbox_repo
+        self.hub_url = hub_url
+
+    def execute(self):
+        deliverable = self.delivery_outbox.get_job()
+        if not deliverable:
+            return
+
+        queue_msg_id, payload = deliverable
+        return self.process(queue_msg_id, payload)
+
+    def process(self, queue_msg_id, job):
+        subscribe_url = job['s']
+        payload = job['payload']
+        attempt = int(job.get('retry', 1))
+
+        try:
+            self._deliver_notification(subscribe_url, payload)
+        except InvalidCallbackResponse as e:
+            logger.exception(e)
+            if attempt < self.MAX_ATTEMPTS:
+                self._retry(subscribe_url, payload, attempt)
+
+        self.delivery_outbox.delete(queue_msg_id)
+
+    def _retry(self, subscribe_url, payload, attempt):
+        logger.info("Delivery failed, re-schedule it")
+        job = {'s': subscribe_url, 'payload': payload, 'retry': attempt + 1}
+        self.delivery_outbox.post_job(job, delay_seconds=self._get_retry_time(attempt))
+
+    def _deliver_notification(self, url, payload):
+        """
+        Send the payload to subscriber's callback url
+
+        https://indieweb.org/How_to_publish_and_consume_WebSub
+        https://www.w3.org/TR/websub/#x7-content-distribution
+        """
+
+        logger.info(f"Sending WebSub payload \n    {payload} to callback URL \n    {url}",)
```

Comment:
with logging it's worth using %s and parameters of the `info` functions (like `logger.info("a %s b", "e")`) - such messages are much easier to filter in the Kibana and Sentry (in case of warnings)